ironfish/src/node.ts

```diff
@@ -226,13 +234,35 @@ export class IronfishNode {
     }
     const workerPool = new WorkerPool({ metrics, numWorkers: workers })
 
-    const consensus = new TestnetParameters()
+    metrics = metrics || new MetricsMonitor({ logger })
+
+    let networkDefinitionJSON = ''
+    // Try fetching custom network definition first, if it exists
+    if (config.get('customNetwork') !== '') {
+      networkDefinitionJSON = await files.readFile(files.resolve(config.get('customNetwork')))
+    } else if (config.get('networkId') === 0) {
+      networkDefinitionJSON = TESTING
+    } else if (config.get('networkId') === 1) {
+      networkDefinitionJSON = TESTNET_PHASE_2
+    }
+
+    const networkDefinition = IJSON.parse(networkDefinitionJSON) as {
+      id: number
+      bootstrapNodes: string[]
+      genesis: SerializedBlock
+      consensus: ConsensusParameters
+    }
+
+    const bootstrapNodes =
+      config.get('bootstrapNodes').length === 0
```

Comment:
We'll probably need to add an `isSet` function to `config` for this. Right now, the only way to start a node up with 0 bootstrap nodes is by passing an empty list to the config flag, and that wouldn't work with this check.

Reply:
Added `isBootstrapNodesSet` to `config`